actionview/lib/action_view/helpers/asset_tag_helper.rb

```diff
@@ -364,6 +375,16 @@ def multiple_sources_tag_builder(type, sources)
           end
         end
 
+        def resolve_image_source(source, skip_pipeline)
+          if source.is_a?(Symbol) || source.is_a?(String)
```

Comment:
What if user passes an object that usually can be resolved into url, but right now can't? E.g. `user.avatar` when no avatar is attached. It will say `Attachment is not a model or has a resolve`, which is misleading.

My version will raise `Can't resolve image into URL: to_model delegated to attachment, but attachment is nil` which, I think, gives the right clue.